source/gui/settingsDialogs.py

```diff
@@ -1190,7 +1205,19 @@ def onOk(self,evt):
 		config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()
 		config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()
 		config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()
-		config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()
+		choice = self.lineIndentationCombo.GetSelection()
+		if choice == 0:
+			config.conf["documentFormatting"]["reportLineIndentation"] = False
+			config.conf["documentFormatting"]["reportToneIndentation"] = False
+		elif choice == 1:
+			config.conf["documentFormatting"]["reportLineIndentation"] = True
+			config.conf["documentFormatting"]["reportToneIndentation"] = False
+		elif choice == 2:
+			config.conf["documentFormatting"]["reportLineIndentation"] = False
+			config.conf["documentFormatting"]["reportToneIndentation"] = True
+		if choice == 3:
+			config.conf["documentFormatting"]["reportLineIndentation"] = True
+			config.conf["documentFormatting"]["reportToneIndentation"] = True

```

Comment:
HMMMMMMMMMMMMMM?


On 7/2/2016 3:36 AM, Brian1Gaff wrote:
>
> In source/gui/settingsDialogs.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r69375799>:
>
> > @@ -1190,7 +1205,19 @@ def onOk(self,evt):
> >  		config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()
> >  		config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()
> >  		config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()
> > -		config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()
> > +		choice = self.lineIndentationCombo.GetSelection()
> > +		if choice == 0:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = False
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = False
> > +		elif choice == 1:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = True
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = False
> > +		elif choice == 2:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = False
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = True
> > +		if choice == 3:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = True
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = True
> Blimey, my Boolean brain just exploded.... Brian 
> bglists@blueyonder.co.uk Sent via blueyonder. Please address personal 
> email to:- briang1@blueyonder.co.uk, putting 'Brian Gaff' in the 
> display name field. ----- Original Message ----- From: "James Teh" 
> <notifications@github.com> To: "nvaccess/nvda" 
> <nvda@noreply.github.com> Sent: Friday, July 01, 2016 11:21 PM 
> Subject: Re: [nvaccess/nvda] NVDA now has the ability to report 
> indents as tones, speech, or both. (#6057 
> <https://github.com/nvaccess/nvda/pull/6057>)
> … <#>
> > @@ -1190,7 +1205,19 @@ def onOk(self,evt): > > 
> config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()> 
> > 
> config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()> 
> > 
> config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()> 
> - > 
> config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()> 
> + choice = self.lineIndentationCombo.GetSelection() > + if choice == 
> 0: > + config.conf["documentFormatting"]["reportLineIndentation"] = 
> False > + config.conf["documentFormatting"]["reportToneIndentation"] = 
> False > + elif choice == 1: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = True > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = False > + 
> elif choice == 2: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = False > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = True > + 
> if choice == 3: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = True > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = True You 
> can simplify this down to two lines: ``` 
> config.conf["documentFormatting"]["reportLineIndentation"] = choice in 
> (1, 3) config.conf["documentFormatting"]["reportToneIndentation"] = 
> choice in (2, 3) ``` --- You are receiving this because you are 
> subscribed to this thread. Reply to this email directly or view it on 
> GitHub: 
> https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69362189 
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69375799>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivaZZCGX0BgvgOEakh0fLY9CYXPhrks5qRjELgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194